std/array.d

```diff
@@ -2268,7 +2268,14 @@ if (isInputRange!Range &&
             copy(stuff, retval[from .. from + stuff.length]);
 
         retval[from + stuff.length .. $] = subject[to .. $];
-        return cast(T[]) retval;
+        static if (is(T == const) || is(T == immutable))
```

Comment:
Should this also check for `is(T == shared)`?

Reply:
I'm not sure. I'm hesitant to do so because who really knows what it means to cast a non-shared array to a shared array (the language spec sure doesn't). Line 2264 should probably be allocated differently to be shared at conception in that case.